components/core/src/clp/Grep.cpp

```diff
@@ -1063,4 +1015,461 @@ size_t Grep::search(Query const& query, size_t limit, Archive& archive, File& co
 
     return num_matches;
 }
+
+vector<QueryLogtype>
+Grep::generate_query_substring_logtypes(string& processed_search_string, ByteLexer& lexer) {
```

Comment:
Given how many cases this method handles and the complexity of the method, we should really have some unit tests for all the cases.